VMware.PSDesiredStateConfiguration.md

```diff
@@ -0,0 +1,147 @@
+
+## Requirements
+
+The following table describes the required dependencies for running VMware.vSphereDSC Resources.
+
+ **Required dependency**   | **Minimum version**
+-------------------------- | -------------------
+`PowerShell`               | 5.1 or 7.0
+
+This module also has additional requirements depending on which version of PowerShell you wish to use it with.
+
+#### PowerShell 7.0 requirements
+---
+**VMware.PSDesiredStateConfiguration** uses the **Invoke-DscResource** cmdlet from **PSDesiredStateConfiguration** module to process individual DSC Resources. In **PowerShell 7.0** the **Invoke-DscResource** cmdlet is a experimental feature.
+The Invoke-DscResource cmdlet and it must be enabled with the following command.
+```
+Enable-ExperimentalFeature PSDesiredStateConfiguration.InvokeDscResource
+```
+ After executing the command you must restart your powershell session. The command needs to be run only once and the cmdlet will be usable.
+ To check if the Invoke-DscResource cmdlet has been enabled correctly run the following command
+```
+Get-Module 'PSDesiredStateConfiguration' -ListAvailable | Select-Object -ExpandProperty ExportedCommands
+```
+##### Important!
+When using **PowerShell 7.0** please keep in mind the list of [Known Limitations](https://github.com/vmware/dscr-for-vmware/blob/master/LIMITATIONS.md)
+
+---
+#### PowerShell 5.1 requirements
+For **PowerShell 5.1** you need to enable Windows Remote Management
+For information on how to enable it read here: [WinRM guide](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management)
+
+
+## Installations
+
+### Manual Installation
+
+1. Copy the VMware.PSDesiredStateConfiguration Module to one of the system PowerShell module directories. For more information on installing PowerShell Modules, please visit [Installing a PowerShell Module](https://docs.microsoft.com/en-us/powershell/scripting/developer/module/installing-a-powershell-module?view=powershell-7).
+2. In PowerShell import the VMware.PSDesiredStateConfiguration Module:
+   ```
+    Import-Module -Name 'VMware.PSDesiredStateConfiguration'
+   ```
+
+   To check if the module was successfully installed:
+   ```
+    Get-Module -Name 'VMware.PSDesiredStateConfiguration'
+   ```
+
+## Use Overview
+VMware.PSDesiredStateConfiguration works with ps1 configuration files and the general flow is as follow:
+- Compile ps1 DSC configuration and transforms given configuration to a configuration object. The object contains the name of the configuration and an array of sorted DSC Resource by dependencies. Each Resource contains **InstanceName** for name of the resource given in the configuration, **ResourceType** which display the type of resource,  **ModuleName** which displays the module of the resource, **Property** which is a hashtable of properties which are unique to the given resource. Example Configuration object from a given configuration:
```

Comment:
Compiles a DSC configuration defined in a PowerShell script file (**ps1**) and transforms the given DSC configuration to a configuration object. The object contains the name of the configuration and an array of DSC Resources. Each DSC Resource object contains an **InstanceName** for the name of the DSC resource given in the configuration, a **ResourceType** which display the type of the DSC Resource, a **ModuleName** which displays the name of the module of the DSC Resource and a **Property** which is a hashtable of properties which are unique for a given DSC Resource. Example Configuration object from a given configuration: